### PR TITLE
czech league added

### PR DESCRIPTION
**Pull request by osmeras**

### CODE REVIEW

File: frontend/src/viewer/logic/score/league/czech.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { Measure } from '../measure';
+import { Score, scoreCircuits, scoreOpenDistance, scoreTriangles } from '../scorer';
+
+abstract class CZXCBase {
+  score(measure: Measure): Score[] {
+    return [
+      ...scoreOpenDistance(measure, 3, this.openDistanceMultiplier),
```
Comment (vicb) [inline frontend/src/viewer/logic/score/league/czech.ts:7]: Make `this.openDistanceMultiplier` a constant ? 
Or at least implement it only once here in the base class

File: frontend/src/viewer/logic/score/league/czech.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  protected abstract flatTriangleMultiplier(): number;
+  protected abstract openDistanceMultiplier(): number;
+
+}
+
+
+export class CzechLocal extends CZXCBase {
+  name = 'Czech (ČPP local)';
+
+  faiTriangleMultiplier(): number {
```
Comment (vicb) [inline frontend/src/viewer/logic/score/league/czech.ts:24]: Could you please add the missing `protected` everywhere ?

File: frontend/src/viewer/logic/score/league/leagues.ts [hunk 3]
```diff
@@ -13,4 +14,7 @@ export const LEAGUES: { [name: string]: League } = {
   ukn: new UKXCLNational(),
   xc: new XContest(),
   wxc: new WXC(),
+  czl: new CzechLocal(),
```
Comment (vicb) [inline frontend/src/viewer/logic/score/league/leagues.ts:17]: Could you please order alphabetically ?

**Review by vicb — CHANGES_REQUESTED** [review]
The changes look good to me.
I only added a few comments.

How frequent are Europe and Outside Europe ?
I'm wondering if the National league would not be enough.

What do you think ?

File: frontend/src/viewer/logic/score/league/czech.ts (hunk 1 above)
Comment (osmeras) [reply frontend/src/viewer/logic/score/league/czech.ts:7]: it is not constant, in 2 leagues it is equal to 1, but in one league it is 0.8

File: frontend/src/viewer/logic/score/league/czech.ts (hunk 2 above)
Comment (osmeras) [reply frontend/src/viewer/logic/score/league/czech.ts:24]: will do.

File: frontend/src/viewer/logic/score/league/leagues.ts (hunk 3 above)
Comment (osmeras) [reply frontend/src/viewer/logic/score/league/leagues.ts:17]: OK

**Comment by vicb** [comment]
Thanks !

What's your name (I'll add you to the list of contributors) ? (I can add "osmeras" if you prefer).

Could you please squash the commits into a single one.

If "by testing" you mean view locally:
- run `npm i` in all the folders with a packages json (root, frontend and run),
- run `npm run debug` in the root,
- navigate to `http://localhost:8080`

**Comment by osmeras** [comment]
I have almost no knowledge in TS, so I did only the minimal changes. The best should be to find where the GPS coordinates are and modify the scores accordingly (so only one league with changing scores), but it is outside my expertise (I am Oracle database developer).

**Comment by vicb** [comment]
> The best should be to find where the GPS coordinates are and modify the scores accordingly

That would be a nice to have but it would require more changes. I think having more league is already a great improvement.
Thanks !

**Comment by osmeras** [comment]
> Could you please squash the commits into a single one.
I have no idea how to do it :-)

**Comment by vicb** [comment]
> I have no idea how to do it :-)

No problem, I'll do it.

Just gibe me your name as you would like it to appear in the contributor list.

**Comment by osmeras** [comment]
> What's your name (I'll add you to the list of contributors) ? (I can add "osmeras" if you prefer).

Stanislav Ošmera

happy to help, I'll try to check if I can help with more

**Comment by vicb** [comment]
Thanks a lot, your updates should be online in the next few minutes.

> I'll try to check if I can help with more

There are many things to do. If there is any particular area you are interested in, let me know and we can figure out something to work on.

Have a great day,
Vic

**Pull request closed without merging**